client/java/src/main/java/com/linecorp/centraldogma/client/Watcher.java

```diff
@@ -178,4 +183,33 @@ default void watch(Consumer<? super T> listener) {
         requireNonNull(listener, "listener");
         watch((revision, value) -> listener.accept(value));
     }
+
+    /**
+     * @return A new watcher, that reuses the supplied watcher and applies transformation.
+     * Note: the returned watcher is effectively filtering in a sense that,
+     * its listeners are *not* notified when a change has no effect on the transformed value.
+     */
+    default <U> Watcher<U> newChild(@Nonnull Function<T, U> transformer) {
+        return new TransformingWatcher<>(this, transformer);
+    }
+    /**
+     * @return A new child, which transformation is a JSON path query.
+     */
+    static Watcher<JsonNode> atJsonPath(@Nonnull Watcher<JsonNode> watcher, String jsonPath) {
```

Comment:
Could accept `Iterable<String>` and `String...`

Reply:
I'm not sure what would such a filter return. An aggregate of several Json paths in a single Json object?

Reply:
Ah, I was walking about JSON path, but yeah, accepting a single JSON pointer is easier and has less dependencies. Could you just rename `jsonPath` to `jsonPointer`?

Reply:
👍